From def3f0289e73b018fd6709e66627ff9a515cf8da Mon Sep 17 00:00:00 2001
From: iam92786 <iam92786@gmail.com>
Date: Fri, 22 Jul 2022 18:25:27 +0530
Subject: [PATCH] u-boot customization: Add blink option to gpio u-boot command

---
 cmd/gpio.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/cmd/gpio.c b/cmd/gpio.c
index 53366f36..4dde213f 100644
--- a/cmd/gpio.c
+++ b/cmd/gpio.c
@@ -11,6 +11,7 @@
 #include <errno.h>
 #include <dm.h>
 #include <asm/gpio.h>
+#include<linux/delay.h>
 
 __weak int name_to_gpio(const char *name)
 {
@@ -22,6 +23,7 @@ enum gpio_cmd {
 	GPIOC_SET,
 	GPIOC_CLEAR,
 	GPIOC_TOGGLE,
+	GPIOC_BLINK,
 };
 
 #if defined(CONFIG_DM_GPIO) && !defined(gpio_status)
@@ -170,6 +172,9 @@ static int do_gpio(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 	case 't':
 		sub_cmd = GPIOC_TOGGLE;
 		break;
+	case 'b':
+		sub_cmd = GPIOC_BLINK;
+		break;
 	default:
 		goto show_usage;
 	}
@@ -217,6 +222,17 @@ static int do_gpio(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv[])
 			if (!IS_ERR_VALUE(value))
 				value = !value;
 			break;
+		case GPIOC_BLINK:
+			//value = 0;
+			while(1)
+			{
+				value =!value;
+				gpio_direction_output(gpio, value);
+				mdelay(100);
+				if(ctrlc())
+					return value;
+			}
+			break;
 		default:
 			goto show_usage;
 		}
-- 
2.25.1

